src/pages/resultsView/ResultsViewPage.tsx

```diff
@@ -413,31 +415,39 @@ export default class ResultsViewPage extends React.Component<IResultsViewPagePro
 
     @computed get pageContent(){
         // if studies are complete but we don't have a tab id in route, we need to derive default
-        return (<div>
-            {
-                // if qeury invalid(we only check gene length for now), return error page
-                (this.resultsViewPageStore.isQueryInvalid) && (
-                    <div className="alert alert-danger queryInvalid" style={{marginBottom: "15px"}} role="alert">
-                        <p>
-                            Queries are limited to 100 genes. Please <a href={`mailto:${AppConfig.serverConfig.skin_email_contact}`}>let us know</a> your use case(s) if you need to query more than 100 genes.
-                        </p>
-                    </div>
-                )
-            }
-            {
-                (this.resultsViewPageStore.studies.isComplete) && (
-                    <Helmet>
-                        <title>{buildResultsViewPageTitle(this.resultsViewPageStore.hugoGeneSymbols, this.resultsViewPageStore.studies.result)}</title>
-                    </Helmet>
-                )
-            }
-            {(this.resultsViewPageStore.studies.isComplete) && (
+        if (this.resultsViewPageStore.invalidStudyIds.result.length > 0) {
+            return (<ErrorScreen
+                title={'Invalid Query'}
+                body={<p>{`You query contains the following invalid study IDs: ${this.resultsViewPageStore.invalidStudyIds.result.join(", ")}`}</p>}
+            />);
+        } else {
+            return (<div>
+                {
+                    // if query invalid(we only check gene length for now), return error page
+                    (this.resultsViewPageStore.isQueryInvalid) && (
```

Comment:
probably we should combine the handling of "invalid" queries here with the above

Reply:
should this be an else if branch?